### ISSUE TITLE
TypeScript Bug: error when defining required array

### ISSUE THREAD

**Issue by rattrayalex** [Automattic/mongoose#10674]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

[TypeScript Playground Link](https://www.typescriptlang.org/play?#code/LAKAlgtgDg9gTgFwAQRgOwOYxgZwKYA0SA3kgLIwAmeANkQMoDGAFnhAIZIC+SAZnDAhIARKkzZ8wgNxIA9LKQA3AKwA6AIwBmVQA5QeAB6xESBAE8oeJADFsSALwlQSJACN2cAFxIcCOGExnHxY8SgBXGjxvYiCXSnYEKJ8-AIwpWKRmGDC4HG80MIhXPDh0kBcuAG0AXTKuMtBzSxtsCmoaB3IqWgAeWxgAPgaQRnRfFpgmVg5OtDwAdyQptnY+7CJ+ttoN7AGACiCY8pc3D2jTCyT6FMxuAgycEPDI6IyXJqTKt5Ojk7+T+KJc4fbzXfyYIhwPAARzCYChlG8fjCVi492O-xOWRyeRIF0s3gAcoVinBITC4QikXAUXdvhV0Zjqoz-lDYfDQtSUSyGUEuKAAJSgIA)

```ts
import mongoose, { Model, Schema } from "mongoose"; // v5.13.8
export type Foo = {
  bar: string
  schedule: {
    date: string;
    hours: number;
  }[];
};

type FooModel = Model<Foo>;

const FooSchema = new Schema<Foo, FooModel, Foo>(
  {
    bar: { type: String },
    schedule: {
      type: [
        {
          date: { type: String, required: true },
          hours: { type: Number, required: true },
        },
      ],
      required: true,
    },
  }
)
```

Error: 
```
        Type '{ date: { type: StringConstructor; required: true; }; hours: { type: NumberConstructor; required: true; }; }' is not assignable to type 'Schema<{ date: string; hours: number; }, Model<{ date: string; hours: number; }, any, any>, {}>'.
          Object literal may only specify known properties, and 'date' does not exist in type 'Schema<{ date: string; hours: number; }, Model<{ date: string; hours: number; }, any, any>, {}>'.
```

(easier to see on the TS playground link, above).

If I change my schema to this, it works fine: 
```ts
    schedule: [
        {
          date: { type: String, required: true },
          hours: { type: Number, required: true },
        },
      ],
```

but of course, I _want_ the `schedule` to be required!

**What is the expected behavior?**

No type error

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**

```json
{
"typescript": "^4.3.5",
"mongodb": "^3.5.7",
"mongoose": "5.13.8",
"@types/mongodb": "3.6.20"
}
```

**Comment by vkarpov15** [comment]
I can confirm this was fixed in v6.0.4 with #10605. We'll move this to our v5.13.10 milestone so we know to backport it when we have time, but I don't expect us to get to that for another week or two. Are you able to upgrade to Mongoose 6.x or is there something blocking you?

**Comment by rattrayalex** [comment]
Awesome thank you!

No known blockers, just general fear of breaking changes and upgrades (and frankly while the migration guide looks comprehensive, it's a little hard to quickly glean what issues we should mostly look out for). But we can definitely give it a try!
